### DISCUSSION TITLE
Is there a way of retrieving live draft data during a draft?

### DISCUSSION THREAD

**Discussion by nmadhavani18** [cwendt94/espn-api#345] [Q&A]
I know there is a way to look at the draft information after everyone has drafted, but is there a way of accessing each draft pick as soon as it has been made? Thanks!

**Comment by cwendt94** [comment]
I have looked at this is the past and I don't believe there is an easy way. However I will try to look into it this year to see if they have change how they do live drafting.

**Comment by nmadhavani18** [comment]
Thanks! I know FantasyPros has figured it out, but haven’t found anything
on it yet.

**Reply by cwendt94** [reply]
ESPN uses websockets for drafting. I was playing around in a mock draft and was able to hook into the draft and see some of the picks and messages being sent. However when I connected to the url it kicked my session on the web browser. So it seems I can't use the same id to connect into the draft stream.

I am going to play around with it some more and hopefully can figure out a solution.

**Comment by nmadhavani18** [comment]
Awesome, let me know if you figure anything out. I’m not super familiar
with websockets.

**Comment by pete7863** [comment]
@nmadhavani18 or @cwendt94 - Were either of you able to find anything more here? Was trying to accomplish myself to no avail, but figured I would ask.

**Reply by cwendt94** [reply]
Not really, I have not found a way to hook into the draft without kicking out the other device that is actually doing the draft.

@nmadhavani18 @pete7863 how were you thinking of using this feature if you could see the live draft data? Trying to figure out if we can hook in, how to show the data.

**Reply by kmcentush** [reply]
I would also like this ability, but I don't have much advice as to how to achieve it technically... Seems like ESPN doesn't update the overall league data until the draft concludes - so no normal endpoints return what we need. Kind of surprising IMO. I imagine there's some internal endpoint that has handles the live draft and it only does a final write when it's complete...

But perhaps that internal endpoint is discoverable by watching the network during a draft? I guess that's similar to what you found with websockets...

**Reply by kmcentush** [reply]
A random thread suggested that you can make another account, assign it as the co-owner, than use that account to avoid the duplicate connection over the same websocket... Seems quite hacky but might work.

Or draft on another device but have the websocket owned by the API client.

**Reply by pete7863** [reply]
I have a draft tool that I use personally and end up manually entering all of the picks. It would be a large improvement if the user did not need to enter these and could just focus on the rest of the data.

**Reply by cwendt94** [reply]
Yeah like @kmcentush it’s hard when the data is only passed through the websockets and then updated right at the end.

**Reply by MuyiLin1** [reply]
you want to share the tool that you be using for your fantasy drafts

**Comment by nmadhavani18** [comment]
I was thinking of making a draft tool that can draft independently.
